golang/server/middleware.go

```diff
@@ -0,0 +1,115 @@
+package server
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/karlkeefer/pngr/golang/errors"
+	"github.com/karlkeefer/pngr/golang/models"
+	"github.com/karlkeefer/pngr/golang/server/jwt"
+	"github.com/karlkeefer/pngr/golang/server/write"
+	"github.com/karlkeefer/pngr/golang/utils"
+)
+
+// wrap does all the middleware together, and parses the user cookie if it's not a session route
+func (srv *server) wrap(h srvHandler) http.HandlerFunc {
```

Comment:
maybe should be a separated into `withUser` and `withEnv` middlewares.
maybe our other middlewares should use `srvHandler` type, too...

🤔